artemis-server/src/main/java/org/apache/activemq/artemis/spi/core/security/ActiveMQJAASSecurityManager.java

```diff
@@ -141,8 +141,10 @@ private Subject getAuthenticatedSubject(final String user,
          try {
             lc.login();
          } catch (LoginException e) {
-            Throwable rootCause = ExceptionUtil.getRootCause(e);
+            Throwable rootCause = ExceptionUtils.getRootCause(e);
             if (rootCause instanceof NoCacheLoginException) {
+               // preserve the original exception for logging
+               rootCause.initCause(e);
```

Comment:
Are we inverting exception chain items on purpose?

Reply:
Could the inversion be avoided if the `SecurityStoreImpl` would use `ExceptionUtils.getRootCause(e)` to check that the root cause is `NoCacheLoginException`?

Reply:
I'm going to rework this. It's not intuitive the way it's currently written.

Reply:
Refactor done.